### DISCUSSION TITLE
How to debug the code

### DISCUSSION THREAD

**Discussion by geekpaolo** [arendst/Tasmota#15357] [Q&A] [answered]
I'm relatively new on Tasmota with VSC+ PlatoformIO

How I can debug the code? Does is possible to insert a sort of "print" in the code and see the results on logs?

Thanks in advance

**Comment by ascillato** [comment, marked as answer]
Hi, What is the issue you are having?

**Reply by sfromis** [reply]
If you browse various Tasmota drivers, you'll see many instances of debugging messages printed, to show up depending on the selected log level.

**Reply by geekpaolo** [reply]
Thanks sfromis!

**Reply by geekpaolo** [reply]
> If you browse various Tasmota drivers, you'll see many instances of debugging messages printed, to show up depending on the selected log level.

Thanks sfromis
